app/src/main/java/net/squanchy/schedule/view/ScheduleDayPageView.kt

```diff
@@ -52,3 +57,19 @@ class ScheduleDayPageView @JvmOverloads constructor(
         override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int) = oldEvents[oldItemPosition] == newEvents[newItemPosition]
     }
 }
+
+private class SnappingLinearLayoutManager(context: Context) : LinearLayoutManager(context) {
+
+    override fun smoothScrollToPosition(recyclerView: RecyclerView, state: RecyclerView.State?, position: Int) {
+        val smoothScroller = TopSnappedSmoothScroller(this, recyclerView.context)
+        smoothScroller.targetPosition = position
+        startSmoothScroll(smoothScroller)
+    }
+}
+
+private class TopSnappedSmoothScroller(private val layoutManager: SnappingLinearLayoutManager, context: Context) : LinearSmoothScroller(context) {
```

Comment:
Does `layoutManager` need to be a `SnappingLinearLayoutManager`? This doesn't seem to use anything specific that isn't in `LayoutManager`, so I'd apply LSP